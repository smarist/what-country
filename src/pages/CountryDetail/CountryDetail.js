import { Box, Grid, Typography } from '@material-ui/core';
import React, { useEffect } from 'react';
import { useParams } from 'react-router';
import { Link } from 'react-router-dom';
import Header from '../../components/Header/Header';
import useStyles from './CountryDetail.style';
import countryServices from '../../Services/country.services';

function CountryDetail({ country, setCountryNameList }) {
  const classes = useStyles();
  const { nameId } = useParams();
  useEffect(() => {
    countryServices.getCountryNameList(nameId)
      .then((res) => {
        setCountryNameList(res);
      })
      .catch(() => {});
  }, []);
  return (
    <Box className={classes.CountryDetailBox}>
      <Box className={classes.headerBox}>
        <Header />
      </Box>
      <Box className={classes.mainBox}>
        <Link to="/Home" className={classes.link}>
          <Box className={classes.backBox}>
            <Box className={classes.arrow}>&larr;</Box>
            <Box className={classes.back}>Back</Box>
          </Box>
        </Link>
        {country?.map(({
          name, capital, subregion, region, population,
          flags, languages, borders, currencies, tld,
          timezones,
        }) => (
          <Grid container>
            <Grid
              item
              xs={12}
              sm={12}
              md={6}
              lg={6}
              className={classes.gridBox}
            >
              <Box className={classes.flagImage}>
                <img src={flags.svg} alt="flag" className={classes.flag} />
              </Box>
            </Grid>
            <Grid item container xs={12} sm={12} md={6} lg={6} className={classes.gridBox}>
              <Grid item xs={12} sm={12} md={12} lg={12}>
                <Typography>{name?.common}</Typography>
              </Grid>
              <Grid xs={12} sm={12} md={6} lg={6} className={classes.gridBox}>
                <Box>
                  <Box className={classes.typoBox}>
                    <Typography>Native Name:</Typography>
                    {(Object.values(name?.nativeName).map((native) => (
                      <Typography>{native?.common}</Typography>
                    )))}
                  </Box>
                  <Box className={classes.typoBox}>
                    <Typography>Population:</Typography>
                    <Typography className={classes.typo2}>{population}</Typography>
                  </Box>
                  <Box className={classes.typoBox}>
                    <Typography>Region:</Typography>
                    <Typography className={classes.typo2}>{region}</Typography>
                  </Box>
                  <Box className={classes.typoBox}>
                    <Typography>Sub Region:</Typography>
                    <Typography className={classes.typo2}>{subregion}</Typography>
                  </Box>
                  <Box className={classes.typoBox}>
                    <Typography>Capital:</Typography>
                    <Typography className={classes.typo2}>{capital}</Typography>
                  </Box>
                </Box>
              </Grid>
              <Grid xs={12} sm={12} md={6} lg={6} className={classes.gridBox}>
                <Box>
                  <Box className={classes.typoBox}>
                    <Typography>Top Level Domain:</Typography>
                    {(Object.values(tld).map((domain) => (
                      <Typography className={classes.typo2}>{domain}</Typography>
                    )))}
                  </Box>
                  <Box className={classes.typoBox}>
                    <Typography>Currency:</Typography>
                    <Typography className={classes.typo2}>
                      {(Object.values(currencies).map((currency) => (
                        <Box className={classes.typoBox}>
                          <Typography className={classes.typo2}>{currency?.name}</Typography>
                          <Typography>
                            (
                            {currency?.symbol}
                            )
                          </Typography>
                        </Box>
                      )))}
                    </Typography>
                  </Box>
                  <Box className={classes.typoBox}>
                    <Typography>Languages:</Typography>
                    {(Object.values(languages)).map((language) => (
                      <Typography className={classes.typo2}>
                        {language}
                      </Typography>
                    ))}
                  </Box>
                  <Box className={classes.typoBox}>
                    <Typography>Time Zones:</Typography>
                    {(Object.values(timezones)).map((zones) => (
                      <Typography className={classes.typo2}>
                        {zones}
                      </Typography>
                    ))}
                  </Box>
                </Box>
              </Grid>
              <Grid xs={12} sm={12} md={12} lg={12} className={classes.gridBox}>
                <Box className={classes.bordersBox}>
                  {borders?.map((border) => (
                    <Box className={classes.borderBox}>{border}</Box>
                  ))}
                </Box>
              </Grid>
            </Grid>
          </Grid>
        ))}
      </Box>
    </Box>
  );
}

export default CountryDetail;
