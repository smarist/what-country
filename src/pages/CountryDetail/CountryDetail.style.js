import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  countryDetailBox: {
    backgroundColor: theme.palette.bgColor.main,
    height: '100vh',
    paddingBottom: '20px',
    color: theme.palette.textColor.main,
    overflowY: 'hidden',
    '&::-webkit-scrollbar': {
      display: 'none',
    },
  },
  headerBox: {
    overFlow: 'hidden',
    position: 'fixed',
    width: '100%',
    zIndex: 2,
  },
  mainBox: {
    padding: '85px 25px !important',
    height: '90vh',
    backgroundColor: theme.palette.bgColor.main,
    color: theme.palette.textColor.main,
    overflowY: 'scroll',
    '&::-webkit-scrollbar': {
      display: 'none',
    },
  },
  gridBox: {
    marginTop: '40px !important',
  },
  flagImage: {
    height: '270px',
    // width: '100%',
    borderRadius: '3px 3px 0 0',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    boxSizing: 'border-box',
  },
  flag: {
    width: '100%',
    maxWidth: '500px',
    height: '100%',
    objectFit: 'cover',
  },
  typoBox: {
    display: 'flex',
  },
  typo2: {
    marginLeft: '7px !important',
  },
  bordersBox: {
    display: 'flex',
    gap: '8px',
  },
  borderBox: {
    width: '50px',
    height: '45px',
    padding: '2px',
    border: '1px solid grey',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  backBox: {
    display: 'flex',
    border: '1px solid grey',
    justifyContent: 'center',
    alignItems: 'center',
    width: '80px',
    padding: '2px 5px',
    gap: '5px',
    background: theme.palette.bgColor2.main,
    color: theme.palette.textColor.main,
    cursor: 'pointer',
  },
  arrow: {
    fontSize: '30px !important',
    marginTop: '-4px !important',
  },
  back: {
    marginTop: '0',
    fontSize: '19px !important',
  },
  link: {
    textDecoration: 'none',
  },
}));

export default useStyles;
