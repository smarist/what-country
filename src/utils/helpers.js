export function objectToArray(val) {
  const obj = val;
  return Object.entries(obj);
}

let timeoutId;

export function debounceHandler(func, delay) {
  if (timeoutId) clearTimeout(timeoutId);
  timeoutId = setTimeout(func, delay);
}
