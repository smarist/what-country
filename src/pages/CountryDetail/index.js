import { connect } from 'react-redux';
import CountryDetail from './CountryDetail';
import { setCountryNameList } from '../../redux/actions/countryAction';

const mapState = ({ country, app }) => ({
  country: country.countryNameList,
  isLoading: app.isLoading,
});

const mapDispatchToProps = dispatch => ({
  setCountryNameList: (data) => setCountryNameList(data)(dispatch),
});

export default connect(mapState, mapDispatchToProps)(CountryDetail);
