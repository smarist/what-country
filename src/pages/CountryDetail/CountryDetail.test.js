import React from 'react';
// import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import CountryDetail from './CountryDetail';
import { withThemeAndStore } from '../../tests/helpers';

jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ nameId: '' }),
  useNavigate: jest
    .fn()
    .mockReturnValue({ goBack: jest.fn() }),
  useLocation: jest.fn().mockReturnValue({}),
  Link: (propVal) => {
    const LinkNav = 'link-navigation';
    return <LinkNav {...propVal} />;
  },
}));

const props = {
  setCountryNameList: jest.fn(),
  country: [],
};

describe('Testing Country Detail Page', () => {
  it('Should render successfully', () => {
    const { queryByTestId } = render(withThemeAndStore(CountryDetail, props));

    const countryDetail = queryByTestId('country-detail');
    expect(countryDetail).toBeTruthy();
    expect(countryDetail).toMatchSnapshot();
  });
});

// describe('Testing Country Detail Page', () => {
//     it('Should render successfully', () => {
//       const { queryByTestId } = render(
//         withThemeAndStore(<Router>
//           <CountryDetail />
//         </Router>, props),
//       );

//       const countryDetail = queryByTestId('country-detail');
//       expect(countryDetail).toBeTruthy();
//       expect(countryDetail).toMatchSnapshot();
//     });
//   });
