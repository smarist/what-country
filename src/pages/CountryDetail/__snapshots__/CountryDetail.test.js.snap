// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Country Detail Page Should render successfully 1`] = `
<div
  class="MuiBox-root MuiBox-root-15"
  data-testid="country-detail"
>
  <div
    class="MuiBox-root MuiBox-root-16 makeStyles-headerBox-2"
  >
    <div
      class="MuiBox-root MuiBox-root-21 makeStyles-headerBox-17"
    >
      <p
        class="MuiTypography-root makeStyles-headerH1-18 MuiTypography-body1"
      >
        Where in the world?
      </p>
      <div
        class="MuiBox-root MuiBox-root-22 makeStyles-switchBox-19"
      >
        <div
          class="MuiBox-root MuiBox-root-23"
        >
          <svg>
            light mode-icon.svg
          </svg>
        </div>
        <div
          class="MuiBox-root MuiBox-root-24"
        >
          LightMode
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-25 makeStyles-mainBox-3"
  >
    <div
      class="MuiBox-root MuiBox-root-26 makeStyles-backBox-11"
    >
      <div
        class="MuiBox-root MuiBox-root-27 makeStyles-arrow-12"
      >
        ‚Üê
      </div>
      <div
        class="MuiBox-root MuiBox-root-28 makeStyles-back-13"
      >
        Back
      </div>
    </div>
  </div>
</div>
`;
